File: VB/ProgressTrackBar/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors

Namespace ProgressTrackBar
	Partial Public Class Form1
		Inherits Form
		Private prb As ProgressBarControl
		Public Sub New()
            prb = New ProgressBarControl()
            InitializeComponent()
            prb.Position = 50
		End Sub

		Private Sub trackBarControl1_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles trackBarControl1.EditValueChanged
			progressBarControl1.Position = trackBarControl1.Value
			prb.Position = trackBarControl1.Value
		End Sub

		Private Sub checkButton1_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles checkButton1.CheckedChanged
			If checkButton1.Checked Then
				myTrackBar1.Properties.Progress = prb
			Else
				myTrackBar1.Properties.Progress = Nothing
			End If
		End Sub

	End Class
End Namespace

File: CS/ProgressTrackBar/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;

namespace ProgressTrackBar
{
    public partial class Form1 : Form
    {
        ProgressBarControl prb;
        public Form1()
        {
            InitializeComponent();
            prb = new ProgressBarControl();
            prb.Position = 50;
        }

        private void trackBarControl1_EditValueChanged(object sender, EventArgs e)
        {
            progressBarControl1.Position = trackBarControl1.Value;
            prb.Position = trackBarControl1.Value;
        }

        private void checkButton1_CheckedChanged(object sender, EventArgs e)
        {
            if (checkButton1.Checked)
                myTrackBar1.Properties.Progress = prb;
            else
                myTrackBar1.Properties.Progress = null;
        }

    }
}